internal/factories/factories.go

```diff
@@ -31,8 +37,29 @@ type ReceiverFactory interface {
 
 	// CreateDefaultConfig creates the default configuration for the Receiver.
 	CreateDefaultConfig() configmodels.Receiver
+
+	// CustomUnmarshaler returns a custom unmarshaler for the configuration or nil if
+	// there is no need for custom unmarshaling. This is typically used if viper.Unmarshal()
+	// is not sufficient to unmarshal correctly.
+	CustomUnmarshaler() CustomUnmarshaler
+
+	// CreateTraceReceiver creates a trace receiver based on this config.
+	// If the receiver type does not support tracing or if the config is not valid
+	// error will be returned instead (ErrReceiverTypeIsNotSupported).
```

Comment:
perhaps, I missed but I didn't see this error defined

Reply:
Wrong comment, let me fix.

Reply:
Fixed.